### DISCUSSION TITLE
Getting an array of json values with same name?

### DISCUSSION THREAD

**Discussion by nulliousp** [nlohmann/json#2988] [Q&A] [answered]
I want to have an array of values returned and all the values have the same name. I can't seem to find out how to do that.

``` json
{
    "name": "John",
    "name": "Harry"
}
```

From that example I want an array with each of the values but all it does it overwrite the first one!

**Comment by nlohmann** [comment]
There is no way to do so as the library only stores the last parsed object key in case of duplicates.

Note that this behavior is compliant to the RFC (https://datatracker.ietf.org/doc/html/rfc8259#section-4):

>  An object whose names are all unique is interoperable in the sense
>  that all software implementations receiving that object will agree on
>  the name-value mappings.  When the names within an object are not
>  unique, the behavior of software that receives such an object is
>  unpredictable.  Many implementations report the last name/value pair
>  only.  Other implementations report an error or fail to parse the
>  object, and some implementations report all of the name/value pairs,
>  including duplicates.

**Comment by nulliousp** [comment, marked as answer]
When I loop through elements, it only shows the value. How would I get the name of the value? for eg
``` json
{
    "person": {
        "name": "John"
    }
}
```

How would I make it give me the value name, which is, "name"?

**Reply by nlohmann** [reply]
See https://json.nlohmann.me/api/basic_json/items/.

**Reply by nulliousp** [reply]
Thanks!

**Reply by nulliousp** [reply]
could you make it an answer so I can mark it as an answer?
